core/chains/evm/client/client.go

```diff
@@ -278,12 +276,18 @@ func (client *client) SendTransaction(ctx context.Context, tx *types.Transaction
 	return client.pool.SendTransaction(ctx, tx)
 }
 
+func (client *client) SimulateTransaction(ctx context.Context, tx *types.Transaction) error {
+	// todo: implement if used
+	return nil
```

Comment:
Will this be filled in for this PR? If not, let's return an explicit error, for example:
```suggestion
	// todo: implement if used
	return errors.New("SimulateTransaction unimplemented")
```